### PR TITLE
feat(cow-hooks): Use new keccak256 based hashing for appData

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

This PR changes how we handle the `appData`

So far, the Frontend was responsible of creating the documents, hashing them (getting the deterministic appData hash that would result from uploading to IPFS), and uploading them (with re-attempt logic, and persistence in local storage to allow temporal network issues)

The changes are related to the PR https://github.com/cowprotocol/services/pull/1671 

As a summary, now we:
* Pass the app as env vars not the default hash that should be used, but the full JSON content 
* The app will maintain most of the appData logic (we keep attaching the same info)
* When is time to sign, what we sign is the raw order, and in place of the `appData` we will use the `keccak256` of the UTF bytes of the JSON string we intend to send to the backend API
* We send the signed order to the backend API, this time with the signature we got before, but in the `appData` we include the full JSON string so the backend can do the check that the signature matches the raw order of the  `keccak256`  of this string

image

## not included
To be done in follow ups

- Explorer
- SDK? 
- appData?

# To Test
This changes ALL FLOWS, so we should make sure:
- Eth Flow works
- Presign flow works
- Swap works
- Limit Order Works
- TWAP works

### CODE REVIEW

**Review by alfetopito — COMMENTED** [review]
(couldn't finish the review yet, will continue later)

**Review by elena-zh — APPROVED** [review]
Hey @anxolin , all order types are successfully created!
However, as you mentioned, Appdata details can't be opened on Explorer (requires implementation)

File: .env [hunk 1]
```diff
@@ -5,14 +5,14 @@ REACT_APP_PRICE_FEED_1INCH_ENABLED=true
...
-REACT_APP_DATA_HASH_DEVELOPMENT=0x828569F802B7F8957F76996BDD875674821E41A688541A9E9EC97D5E897D44A7
-REACT_APP_DATA_HASH_LOCAL=0xA5DAE7A114F1BD6BB9B3FF976150380A95CB18856212DB555C25EF9D7801E9A4
+# AppData, build yours at https://explorer.cow.fi/appdata
+REACT_APP_FULL_APP_DATA_PRODUCTION='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"production"}}'
+REACT_APP_FULL_APP_DATA_ENS='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"ens"}}'
+REACT_APP_FULL_APP_DATA_BARN='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"barn"}}'
+REACT_APP_FULL_APP_DATA_STAGING='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"staging"}}'
+REACT_APP_FULL_APP_DATA_PR='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"pr"}}'
+REACT_APP_FULL_APP_DATA_DEVELOPMENT='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"development"}}'
+REACT_APP_FULL_APP_DATA_LOCAL='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"local"}}'
```
[the remark this answers is not in the document]
Comment (anxolin) [reply .env:15]: I will update, sure, I guess this one is still compatible

File: .env.production [hunk 2]
```diff
@@ -5,14 +5,14 @@ REACT_APP_PRICE_FEED_1INCH_ENABLED=true
...
-REACT_APP_DATA_HASH_DEVELOPMENT=0x828569F802B7F8957F76996BDD875674821E41A688541A9E9EC97D5E897D44A7
-REACT_APP_DATA_HASH_LOCAL=0xA5DAE7A114F1BD6BB9B3FF976150380A95CB18856212DB555C25EF9D7801E9A4
+# AppData, build yours at https://explorer.cow.fi/appdata
+REACT_APP_FULL_APP_DATA_PRODUCTION='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"production"}}'
+REACT_APP_FULL_APP_DATA_ENS='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"ens"}}'
+REACT_APP_FULL_APP_DATA_BARN='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"barn"}}'
+REACT_APP_FULL_APP_DATA_STAGING='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"staging"}}'
+REACT_APP_FULL_APP_DATA_PR='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"pr"}}'
+REACT_APP_FULL_APP_DATA_DEVELOPMENT='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"development"}}'
+REACT_APP_FULL_APP_DATA_LOCAL='{"version":"0.1.0","appCode":"CowSwap","metadata":{"environment":"local"}}'
```
[the remark this answers is not in the document]
Comment (anxolin) [reply .env.production:15]: true, our .env is not super in shape

I will delete

**Comment by anxolin** [comment]
I will wait a bit to merge until we have the matching explorer PR

File: .env (hunk 1 above)
[the remark this answers is not in the document]
Comment (anxolin) [reply .env:15]: mm, it was not, using now `REACT_APP_FULL_APP_DATA_PRODUCTION='{"version":"0.7.0","appCode":"CoW Swap","environment":"production","metadata":{}}'`

File: .env [hunk 3]
```diff
@@ -43,12 +43,12 @@ REACT_APP_FORTMATIC_SITE_VERIFICATION="LzjrtdM7hqVJfvvA"
...
-REACT_APP_DOMAIN_REGEX_PR="^pr\d+--cowswap\.review|^(swap-dev-git-[\w\d-]+|swap-\w{9}-)cowswap\.vercel\.app"
-REACT_APP_DOMAIN_REGEX_DEV="^(cowswap\.dev|dev\.swap\.cow\.fi|swap-develop\.vercel\.app)"
-REACT_APP_DOMAIN_REGEX_STAGING="^(cowswap\.staging|staging\.swap\.cow\.fi|swap-staging\.vercel\.app)"
+REACT_APP_DOMAIN_REGEX_PR="^pr\d+--CoW Swap\.review|^(swap-dev-git-[\w\d-]+|swap-\w{9}-)CoW Swap\.vercel\.app"
+REACT_APP_DOMAIN_REGEX_DEV="^(CoW Swap\.dev|dev\.swap\.cow\.fi|swap-develop\.vercel\.app)"
+REACT_APP_DOMAIN_REGEX_STAGING="^(CoW Swap\.staging|staging\.swap\.cow\.fi|swap-staging\.vercel\.app)"
 REACT_APP_DOMAIN_REGEX_PROD="^(swap\.cow\.fi|swap-prod\.vercel\.app)$"
 REACT_APP_DOMAIN_REGEX_BARN="^(barn\.cow\.fi|swap-barn\.vercel\.app)$"
-REACT_APP_DOMAIN_REGEX_ENS="(:?^cowswap\.eth|ipfs)"
+REACT_APP_DOMAIN_REGEX_ENS="(:?^CoW Swap\.eth|ipfs)"
```
Comment (alfetopito) [inline .env:51]: Please revert this change.
Those are urls patters

File: src/modules/appData/utils/fullAppData.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+const DEFAULT_FULL_APP_DATA = '{"version":"0.7.0","appCode":"CoW Swap","metadata":{}}'
+
+let appData: AppDataInfo = (() => {
+  const fullAppData = _getFullAppData()
+  return _fromFullAppData(fullAppData)
+})()
+
+export function getAppData() {
+  return appData
+}
```
Comment (alfetopito) [inline src/modules/appData/utils/fullAppData.ts:15]: Nitpick: do you need a fn for this?

File: src/modules/appData/utils/fullAppData.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    default:
+      break
+  }
+  return DEFAULT_FULL_APP_DATA
+}
+
+function _fromFullAppData(fullAppData: string): AppDataInfo {
+  return {
+    doc: JSON.parse(fullAppData),
+    fullAppData: fullAppData,
```
Comment (alfetopito) [inline src/modules/appData/utils/fullAppData.ts:55]: Nitpick:

```suggestion
    fullAppData,
```

**Review by alfetopito — CHANGES_REQUESTED** [review]

File: .env (hunk 3 above)
Comment (anxolin) [reply .env:51]: yep, i must have find and replace by mistake
reverting

**Review by alfetopito — APPROVED** [review]
